### ISSUE TITLE
An unexpected error occured while processing the network profile of the VM. Please retry later.

### ISSUE THREAD

**Issue by sanghoonkim0918** [Azure/azure-cli#12751]
## Describe the bug

**Command Name**
`az vm create`

**Errors:**
```
Deployment failed. Correlation ID: 2ff74f6d-9636-434e-a772-ea3bd265ba7d. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "NetworkingInternalOperationError",
        "message": "An unexpected error occured while processing the network profile of the VM. Please retry later."
      }
    ]
  }
}
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az vm create --subscription {} --location {} --resource-group {} --name {} --image {} --size {} --generate-ssh-keys --output {}`

## Environment Summary
```
Linux-5.0.0-1032-azure-x86_64-with-Ubuntu-18.04-bionic
Python 3.6.9

azure-cli 2.2.0

```
## Additional Context

I get this error quite often.
I think I have this error only when I try to make a VM on eastus2.

**Comment by yonzhan** [comment]
@qwordy please take a look

**Comment by qwordy** [comment]
Is there any existing resource in the resource group? And can you add `--debug` to your command and post the output here?

**Comment by yonzhan** [comment]
@sanghoonkim0918 any feedback/update for above question?

**Comment by JoeSainsburys** [comment]
I have the same issue with north europe (is this a capacity issue?) Fails to create a vmss from terraform. Then I get the same error on start.

```

az vmss start -n build-server -g rg-build --debug
Command arguments: ['vmss', 'start', '-n', 'build-server', '-g', 'rg-build', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f8a47bc1d08>, <function OutputProducer.on_global_arguments at 0x7f8a476edae8>, <function CLIQuery.on_global_arguments at 0x7f8a47715b70>]
Event: CommandInvoker.OnPreCommandTableCreate []
Installed command modules ['acr', 'acs', 'advisor', 'ams', 'apim', 'appconfig', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'deploymentmanager', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'interactive', 'iot', 'iotcentral', 'keyvault', 'kusto', 'lab', 'managedservices', 'maps', 'monitor', 'natgateway', 'netappfiles', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'reservations', 'resource', 'role', 'search', 'security', 'servicebus', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'vm']
Loaded module 'acr' in 0.005 seconds.
Loaded module 'acs' in 0.022 seconds.
Loaded module 'advisor' in 0.002 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'ams' in 0.005 seconds.
Loaded module 'apim' in 0.003 seconds.
Loaded module 'appconfig' in 0.004 seconds.
Loaded module 'appservice' in 0.009 seconds.
Loaded module 'backup' in 0.004 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'batch' in 0.008 seconds.
Loaded module 'batchai' in 0.004 seconds.
Loaded module 'billing' in 0.003 seconds.
Loaded module 'botservice' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'cdn' in 0.003 seconds.
Loaded module 'cloud' in 0.002 seconds.
Loaded module 'cognitiveservices' in 0.002 seconds.
Loaded module 'configure' in 0.001 seconds.
Loaded module 'consumption' in 0.003 seconds.
Loaded module 'container' in 0.004 seconds.
Loaded module 'cosmosdb' in 0.009 seconds.
Loaded module 'deploymentmanager' in 0.003 seconds.
Loaded module 'dla' in 0.005 seconds.
Loaded module 'dls' in 0.004 seconds.
Loaded module 'dms' in 0.003 seconds.
Loaded module 'eventgrid' in 0.003 seconds.
Loaded module 'eventhubs' in 0.005 seconds.
Loaded module 'extension' in 0.002 seconds.
Loaded module 'feedback' in 0.001 seconds.
Loaded module 'find' in 0.001 seconds.
Loaded module 'hdinsight' in 0.002 seconds.
Loaded module 'interactive' in 0.000 seconds.
Loaded module 'iot' in 0.008 seconds.
Loaded module 'iotcentral' in 0.002 seconds.
Loaded module 'keyvault' in 0.006 seconds.
Loaded module 'kusto' in 0.003 seconds.
Loaded module 'lab' in 0.004 seconds.
Loaded module 'managedservices' in 0.002 seconds.
Loaded module 'maps' in 0.002 seconds.
Loaded module 'monitor' in 0.006 seconds.
Loaded module 'natgateway' in 0.002 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'netappfiles' in 0.006 seconds.
Loaded module 'network' in 0.031 seconds.
Loaded module 'policyinsights' in 0.002 seconds.
Loaded module 'privatedns' in 0.005 seconds.
Loaded module 'profile' in 0.002 seconds.
Loaded module 'rdbms' in 0.006 seconds.
Loaded module 'redis' in 0.003 seconds.
Loaded module 'relay' in 0.004 seconds.
Loaded module 'reservations' in 0.003 seconds.
Loaded module 'resource' in 0.007 seconds.
Loaded module 'role' in 0.005 seconds.
Loaded module 'search' in 0.002 seconds.
Loaded module 'security' in 0.003 seconds.
Loaded module 'servicebus' in 0.005 seconds.
Loaded module 'servicefabric' in 0.006 seconds.
Loaded module 'signalr' in 0.002 seconds.
Loaded module 'sql' in 0.008 seconds.
Loaded module 'sqlvm' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'storage' in 0.032 seconds.
Loaded module 'vm' in 0.016 seconds.
Loaded all modules in 0.312 seconds. (note: there's always an overhead with the first module loaded)
Extensions directory: '/home/joe/.azure/cliextensions'
Found 1 extensions: ['aks-preview']
Extensions directory: '/home/joe/.azure/cliextensions'
Extension compatibility result: is_compatible=True cli_core_version=2.0.81 min_required=2.0.49 max_required=None
Event: CommandLoader.OnLoadCommandTable []
Extensions directory: '/home/joe/.azure/cliextensions'
Loaded extension 'aks-preview' in 0.014 seconds.
Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f8a46ed0598>]
az_command_data_logger : command args: vmss start -n {} -g {} --debug
metadata file logging enabled - writing logs to '/home/joe/.azure/commands'.
Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f8a46e6c2f0>]
Event: CommandInvoker.OnPostArgumentLoad []
Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f8a46e9ef28>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7f8a46e2f158>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x7f8a44fa0598>]
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f8a476edb70>, <function CLIQuery.handle_query_parameter at 0x7f8a47715bf8>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f8a46e2f048>, <function handler at 0x7f8a44e6e598>]
Getting management service client client_type=ComputeManagementClient
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
msrest.service_client : Accept header absent and forced to application/json
attempting to read file /home/joe/.azure/accessTokens.json as utf-8-sig
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - Authority:Performing instance discovery: ...
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - Authority:Performing static instance discovery
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - Authority:Authority validated via static instance discovery
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - TokenRequest:Getting token from cache with refresh if necessary.
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - CacheDriver:Found 17 potential entries.
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - CacheDriver:Resource specific token found.
adal-python : 8b2d13a0-3c73-40c7-91c9-ead9bdf78df2 - CacheDriver:Returning token from cache lookup, AccessTokenId: *removed*, RefreshTokenId: *removed*
msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/*removed*/resourceGroups/rg-build/providers/Microsoft.Compute/virtualMachineScaleSets/build-server/start?api-version=2019-07-01'
msrest.http_logger : Request method: 'POST'
msrest.http_logger : Request headers:
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'User-Agent': 'python/3.6.5 (Linux-4.4.0-18362-Microsoft-x86_64-with-debian-buster-sid) msrest/0.6.9 msrest_azure/0.6.2 azure-mgmt-compute/10.0.0 Azure-SDK-For-Python AZURECLI/2.0.81'
msrest.http_logger : Request body:
msrest.http_logger : None
msrest.universal_http : Configuring redirects: allow=True, max=30
msrest.universal_http : Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http : Configuring proxies: ''
msrest.universal_http : Evaluate proxies against ENV settings: True
urllib3.connectionpool : Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool : https://management.azure.com:443 "POST /subscriptions/*removed*/resourceGroups/rg-build/providers/Microsoft.Compute/virtualMachineScaleSets/build-server/start?api-version=2019-07-01 HTTP/1.1" 202 0
msrest.http_logger : Response status: 202
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'Location': 'https://management.azure.com/subscriptions/*removed*/providers/Microsoft.Compute/locations/northeurope/operations/e6549e75-f72f-4883-a75a-4455c12f20c5?monitor=true&api-version=2019-07-01'
msrest.http_logger :     'Azure-AsyncOperation': 'https://management.azure.com/subscriptions/*removed*/providers/Microsoft.Compute/locations/northeurope/operations/e6549e75-f72f-4883-a75a-4455c12f20c5?api-version=2019-07-01'
msrest.http_logger :     'x-ms-ratelimit-remaining-resource': 'Microsoft.Compute/VMScaleSetActions3Min;238,Microsoft.Compute/VMScaleSetActions30Min;1197,Microsoft.Compute/VMScaleSetBatchedVMRequests5Min;1198,Microsoft.Compute/VmssQueuedVMOperations;4799'
msrest.http_logger :     'x-ms-request-charge': '1'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'x-ms-request-id': 'e6549e75-f72f-4883-a75a-4455c12f20c5'
msrest.http_logger :     'Server': 'Microsoft-HTTPAPI/2.0, Microsoft-HTTPAPI/2.0'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-writes': '1199'
msrest.http_logger :     'x-ms-correlation-request-id': '198d75e4-8ca4-4153-8182-ec2d78b753ff'
msrest.http_logger :     'x-ms-routing-request-id': 'UKSOUTH:20200427T112034Z:198d75e4-8ca4-4153-8182-ec2d78b753ff'
msrest.http_logger :     'X-Content-Type-Options': 'nosniff'
msrest.http_logger :     'Date': 'Mon, 27 Apr 2020 11:20:34 GMT'
msrest.http_logger :     'Content-Length': '0'
msrest.http_logger : Response content:
msrest.http_logger :
msrest.service_client : Accept header absent and forced to application/json
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - Authority:Performing instance discovery: ...
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - Authority:Performing static instance discovery
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - Authority:Authority validated via static instance discovery
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - TokenRequest:Getting token from cache with refresh if necessary.
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - CacheDriver:Found 17 potential entries.
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - CacheDriver:Resource specific token found.
adal-python : 64ff5a17-c662-434d-b58f-f15fcc7033ce - CacheDriver:Returning token from cache lookup, AccessTokenId: *removed*, RefreshTokenId: *removed*
msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/*removed*/providers/Microsoft.Compute/locations/northeurope/operations/e6549e75-f72f-4883-a75a-4455c12f20c5?api-version=2019-07-01'
msrest.http_logger : Request method: 'GET'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'x-ms-client-request-id': '1bd451a8-8879-11ea-a575-34415d898ec8'
msrest.http_logger :     'User-Agent': 'python/3.6.5 (Linux-4.4.0-18362-Microsoft-x86_64-with-debian-buster-sid) msrest/0.6.9 msrest_azure/0.6.2 azure-mgmt-compute/10.0.0 Azure-SDK-For-Python AZURECLI/2.0.81'
msrest.http_logger : Request body:
msrest.http_logger : None
msrest.universal_http : Configuring redirects: allow=True, max=30
msrest.universal_http : Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http : Configuring proxies: ''
msrest.universal_http : Evaluate proxies against ENV settings: True
urllib3.connectionpool : Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool : https://management.azure.com:443 "GET /subscriptions/*removed*/providers/Microsoft.Compute/locations/northeurope/operations/e6549e75-f72f-4883-a75a-4455c12f20c5?api-version=2019-07-01 HTTP/1.1" 200 None
msrest.http_logger : Response status: 200
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Transfer-Encoding': 'chunked'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'Content-Encoding': 'gzip'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'Retry-After': '480'
msrest.http_logger :     'Vary': 'Accept-Encoding'
msrest.http_logger :     'x-ms-ratelimit-remaining-resource': 'Microsoft.Compute/GetOperation3Min;14985,Microsoft.Compute/GetOperation30Min;29967'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'x-ms-request-id': '340b0904-b49e-4504-9a68-61d6e5c92d77'
msrest.http_logger :     'Server': 'Microsoft-HTTPAPI/2.0, Microsoft-HTTPAPI/2.0'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-reads': '11971'
msrest.http_logger :     'x-ms-correlation-request-id': '2f2e5fb7-6d07-4a5a-a436-0591cd06af65'
msrest.http_logger :     'x-ms-routing-request-id': 'UKSOUTH:20200427T112104Z:2f2e5fb7-6d07-4a5a-a436-0591cd06af65'
msrest.http_logger :     'X-Content-Type-Options': 'nosniff'
msrest.http_logger :     'Date': 'Mon, 27 Apr 2020 11:21:04 GMT'
msrest.http_logger : Response content:
msrest.http_logger : {
  "startTime": "2020-04-27T11:20:34.4495245+00:00",
  "endTime": "2020-04-27T11:20:35.3558437+00:00",
  "status": "Failed",
  "error": {
    "code": "NetworkingInternalOperationError",
    "message": "An unexpected error occured while processing the network profile of the VM. Please retry later."
  },
  "name": "e6549e75-f72f-4883-a75a-4455c12f20c5"
}
msrest.exceptions : An unexpected error occured while processing the network profile of the VM. Please retry later.
cli.azure.cli.core.util : Deployment failed. Correlation ID: 2f2e5fb7-6d07-4a5a-a436-0591cd06af65. An unexpected error occured while processing the network profile of the VM. Please retry later.
Deployment failed. Correlation ID: 2f2e5fb7-6d07-4a5a-a436-0591cd06af65. An unexpected error occured while processing the network profile of the VM. Please retry later.
az_command_data_logger : exit code: 1
telemetry.save : Save telemetry record of length 5998 in cache
telemetry.check : Negative: The /home/joe/.azure/telemetry.txt was modified at 2020-04-27 12:16:25.458117, which in less than 600.000000 s

```

**Comment by JoeSainsburys** [comment]
I chased this up with MS and they seem to have found the issue and are in the process of fixing it

Hi Joe,
 
I hope this e-mail finds you well
 
Thank you for your quick response and I am glad that the issue is fixed .
The Microsoft Azure Platform team has investigated the issue which affected the Virtual Machine management operations like start/deallocate/delete. Our investigation determined that the VIPs allocated to these deployments were not released
by backend networking component on time due to a bug in the code which handles the release requests. 
  
We have identified a fix for this issue which will roll out to production over the next 2 weeks following safe deployment process.  Until fully deployed, we are running multiple daily scans to
identify impacted resources and mitigate them proactively. 
 We apologize for any inconvenience that may have been caused by this behavior
Please let me know if you have other questions to continue troubleshooting or mark this case as resolved .

**Comment by qwordy** [comment]
@JoeSainsburys Hi, would you please share what's the root cause and how the issue is solved?

**Comment by houk-ms** [comment]
Closed due to no response for a long time. Open anthoer issue if anthor error occurs.

**Comment by JoeSainsburys** [comment]
@qwordy sorry for the late response. This is all I know. The ticket number was REG:120042825000603 if that helps.

**Comment by houk-ms** [comment]
@sanghoonkim0918 @JoeSainsburys Thanks for reporting this issue. But we could not reproducce the error this time. If you encounter this issue again, feel free to reopen the issue here so that we can see more details.

**Comment by bart986** [comment]
Hello,
Allow me to share similar issue with our VMSS. It is pretty stable and each day it is being started and stopped by runbooks that twe hold in automation account. Until today: without particular reason vmss didn't start instance today, when I checked erorrs in jobs I found: 
![image](https://user-images.githubusercontent.com/68948199/88785204-318cde00-d191-11ea-925e-76bcbb23e423.png)
looks like internal microsoft error, vmss is completly frozen, restarting, stopping, starting each instance takes around 55minutes and fails each time, same when I try to restart vmss service. It is huge P0 situation because this vmss corsponds to aks for our dev environment, would be great it any Azure tech specialist could have a look on that
